src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -88,6 +101,26 @@ public boolean publish(String message, String color) {
         return result;
     }
 
+    private String getTokenToUse() {
+        if (authTokenCredentialId != null && !authTokenCredentialId.isEmpty()) {
+            StringCredentials credentials = lookupCredentials(authTokenCredentialId);
+            if (credentials != null) {
+                logger.info("Using Integration Token Credential ID.");
+                return credentials.getSecret().getPlainText();
+            }
+        }
+
+        logger.info("Using Integration Token.");
+
+        return token;
+    }
+
+    private StringCredentials lookupCredentials(String credentialId) {
+        List<StringCredentials> credentials = CredentialsProvider.lookupCredentials(StringCredentials.class, Jenkins.getInstance(), ACL.SYSTEM, Collections.<DomainRequirement>emptyList());
```

Comment:
The GitHub plugin has credentials integrations in Jenkins global config.
![image](https://cloud.githubusercontent.com/assets/983526/17889272/4d322a80-68fd-11e6-9337-c0486726f613.png)
See the code here: https://github.com/jenkinsci/github-plugin/blob/master/src/main/java/org/jenkinsci/plugins/github/config/GitHubServerConfig.java

Users should be able to select existing credentials from the drop down or create a new one on the fly from the slack config location.